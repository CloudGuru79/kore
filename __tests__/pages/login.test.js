import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Login from '../../pages/login'

const authProvider = {
  metadata: {
    name: 'default'
  },
  spec: {
    displayName: 'My Provider',
    config: {
      myprovider: {}
    }
  }
}

describe('Login', () => {
  it('shows login for auth provider', () => {
    const app = shallow(<Login authProvider={authProvider} connectorId="default-myprovider" />)

    expect(app.find('Card').prop('title')).toEqual('Login')
    expect(app.find('Button a').prop('href')).toEqual('/login/auth?provider=default-myprovider')
    expect(app.find('Button a').text()).toEqual(`Login with ${authProvider.spec.displayName}`)
  })
})

describe('Snapshot testing', () => {
  it('shows login', () => {
    const component = renderer.create(<Login authProvider={authProvider} connectorId="default-myprovider" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})