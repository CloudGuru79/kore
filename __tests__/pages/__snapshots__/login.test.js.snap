// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing shows login not using embedded auth 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "minHeight": 280,
      "padding": 24,
    }
  }
>
  <div
    className="ant-row-flex ant-row-flex-center"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Login
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div>
            <p>
              Login using the configured Identity Provider below
            </p>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              style={
                Object {
                  "margin": "5px",
                }
              }
              type="button"
            >
              <a
                href="/login/auth"
              >
                Login with Identity Provider
              </a>
            </button>
            <button
              className="ant-btn"
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
              type="button"
            >
              <span>
                Local user login
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot testing shows login using embedded auth 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "minHeight": 280,
      "padding": 24,
    }
  }
>
  <div
    className="ant-row-flex ant-row-flex-center"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-card ant-card-bordered"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              Login
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div>
            <p>
              Login using the configured Identity Provider below
            </p>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              style={
                Object {
                  "margin": "5px",
                }
              }
              type="button"
            >
              <a
                href="/login/auth?provider=default-myprovider"
              >
                Login with 
                My Provider
              </a>
            </button>
            <button
              className="ant-btn"
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
              type="button"
            >
              <span>
                Local user login
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
